source/material_model/drucker_prager.cc

```diff
@@ -0,0 +1,396 @@
+                             "in both the density and viscosity formulas. Units: $K$.");
+          prm.declare_entry ("Reference viscosity", "1e22",
+                             Patterns::Double (0),
+                             "The value of the reference viscosity $\\eta_0$. Units: $kg/m/s$.");
+          prm.declare_entry ("Thermal conductivity", "4.7",
+                             Patterns::Double (0),
+                             "The value of the thermal conductivity $k$. "
+                             "Units: $W/m/K$.");
+          prm.declare_entry ("Reference specific heat", "1250",
+                             Patterns::Double (0),
+                             "The value of the specific heat $cp$. "
+                             "Units: $J/kg/K$.");
+          prm.declare_entry ("Thermal expansion coefficient", "2e-5",
+                             Patterns::Double (0),
+                             "The value of the thermal expansion coefficient $\\beta$. "
+                             "Units: $1/K$.");
+          prm.enter_subsection ("Viscosity");
+          {
+            prm.declare_entry ("Minimum viscosity", "1e19",
+                               Patterns::Double (0),
+                               "The value of the minimum viscosity cutoff. Units: $Pas$.");
+            prm.declare_entry ("Maximum viscosity", "1e24",
+                               Patterns::Double (0),
+                               "The value of the maximum viscosity cutoff. Units: $Pas$.");
+            prm.declare_entry ("Reference strain rate", "1e-15",
+                               Patterns::Double (0),
+                               "The value of the initial strain rate prescribed during the "
+                               "first nonlinear iteration. Units: $1/s$.");
+            prm.declare_entry ("Angle internal friction", "0",
+                               Patterns::Double (0),
+                               "Angle of internal friction. For a value of zero, the von Mises "
+                               "criterion is retrieved. Angles higher than 30 degrees are "
+                               "harder to solve numerically. Units: degrees.");
+            prm.declare_entry ("Cohesion", "2e7",
+                               Patterns::Double (0),
+                               "Cohesion. Units: Pa.");
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    DP<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection ("Compositional fields");
+      {
+        // This material model does not incorporate a dependence of material properties
+        // on compositional fields
+        AssertThrow(prm.get_integer ("Number of fields") == 0,
+                    ExcMessage("This material model does not account for compositional fields."));
```

Comment:
Why do you assert this? One could just want to advect along fields that then simply don't enter into the material model.